zapcore/sampler_test.go

```diff
@@ -40,103 +40,97 @@ func fakeSampler(lvl LevelEnabler, tick time.Duration, first, thereafter int) (F
+		require.Equal(t, 1, len(entry.Context), "Unexpected number of fields.")
+		require.Equal(t, lvl, entry.Level, "Unexpected level.")
+		f := entry.Context[0]
+		require.Equal(t, "iter", f.Key, "Unexpected field key.")
+		require.Equal(t, Int64Type, f.Type, "Unexpected field type")
+		seen[i] = f.Integer
 	}
-	return expected
+	assert.Equal(t, seq, seen, "Unexpected sequence logged at level %v.", lvl)
 }
 
-func writeIter(fac Facility, n int, lvl Level) {
+func writeSequence(fac Facility, n int, lvl Level) {
+	// All tests using writeSequence verify that counters are shared between
+	// parent and child facilities.
 	fac = fac.With([]Field{makeInt64Field("iter", n)})
-	if ce := fac.Check(Entry{Level: lvl}, nil); ce != nil {
+	if ce := fac.Check(Entry{Level: lvl, Time: time.Now().UTC()}, nil); ce != nil {
 		ce.Write()
 	}
 }
 
 func TestSampler(t *testing.T) {
-	for _, lvl := range []Level{DebugLevel, InfoLevel, WarnLevel, ErrorLevel, DPanicLevel} {
+	for _, lvl := range []Level{DebugLevel, InfoLevel, WarnLevel, ErrorLevel, DPanicLevel, PanicLevel, FatalLevel} {
 		sampler, logs := fakeSampler(DebugLevel, time.Minute, 2, 3)
+
+		// Ensure that counts aren't shared between levels.
+		probeLevel := DebugLevel
+		if lvl == DebugLevel {
+			probeLevel = InfoLevel
+		}
+		for i := 0; i < 10; i++ {
+			writeSequence(sampler, 1, probeLevel)
+		}
+		// Clear any output.
+		logs.TakeAll()
+
 		for i := 1; i < 10; i++ {
-			writeIter(sampler, i, lvl)
+			writeSequence(sampler, i, lvl)
 		}
-		assert.Equal(t, buildExpectation(lvl, 1, 2, 5, 8), logs.All(), "Unexpected output from sampled logger.")
+		assertSequence(t, logs.TakeAll(), lvl, 1, 2, 5, 8)
 	}
 }
 
 func TestSamplerDisabledLevels(t *testing.T) {
 	sampler, logs := fakeSampler(InfoLevel, time.Minute, 1, 100)
 
 	// Shouldn't be counted, because debug logging isn't enabled.
-	writeIter(sampler, 1, DebugLevel)
-	writeIter(sampler, 2, InfoLevel)
-	expected := buildExpectation(InfoLevel, 2)
-	assert.Equal(t, expected, logs.All(), "Expected to disregard disabled log levels.")
+	writeSequence(sampler, 1, DebugLevel)
+	writeSequence(sampler, 2, InfoLevel)
+	assertSequence(t, logs.TakeAll(), InfoLevel, 2)
 }
 
-func TestSamplerWithSharesCounters(t *testing.T) {
-	sampler, logs := fakeSampler(DebugLevel, time.Minute, 1, 100)
+func TestSamplerTicking(t *testing.T) {
+	// Ensure that we're resetting the sampler's counter every tick.
+	sampler, logs := fakeSampler(DebugLevel, 10*time.Millisecond, 5, 10)
 
-	first := sampler.With([]Field{makeInt64Field("child", 1)})
-	for i := 1; i < 10; i++ {
-		writeIter(first, i, InfoLevel)
-	}
-	second := sampler.With([]Field{makeInt64Field("child", 2)})
-	// The new child logger should share the same counters, so we don't expect to
-	// write these logs.
-	for i := 10; i < 20; i++ {
-		writeIter(second, i, InfoLevel)
+	// If we log five or fewer messages every tick, none of them should be
+	// sampled.
```

Comment:
"sampled" to me means that we see them, to me this should say "none of them should be dropped by the sampler"?